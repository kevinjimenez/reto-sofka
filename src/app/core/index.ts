export { spinnerInterceptor } from './interceptors/spinner.interceptor';
export { productsResolver } from './resolvers/products.resolver';
export { ProductsService } from './services/products.service';
export { SpinnerService } from './services/spinner.service';
