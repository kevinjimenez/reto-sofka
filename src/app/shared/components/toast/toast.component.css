.toast-container {
	position: relative;
  cursor: pointer;
}

.toast-contend {
	border: solid 1px #ccc;
	position: fixed;
	top: 0.5rem;
	right: 0.5rem;
	z-index: 9999;
	padding-inline: 1rem;
	border-radius: 0.3rem;
	padding-bottom: 0.5rem;
	padding-top: 0.5rem;
  box-shadow: rgba(60, 64, 67, 0.3) 0px 1px 2px 0px, rgba(60, 64, 67, 0.15) 0px 2px 6px 2px;
  background: #ced6e0;
}


