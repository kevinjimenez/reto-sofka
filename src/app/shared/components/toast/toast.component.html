@if (isShow()) {
	<a class="toast-container" (click)="hide()">
		<section class="toast-contend">
			{{ text() }}
		</section>
	</a>
}
