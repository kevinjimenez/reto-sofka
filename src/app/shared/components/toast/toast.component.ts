import { Component, Input, input, model, OnInit } from '@angular/core';

@Component({
	selector: 'app-toast',
	standalone: true,
	imports: [],
	templateUrl: './toast.component.html',
	styleUrl: './toast.component.css'
})
export class ToastComponent {
	public isShow = model<boolean>(false);
	// public text = input<string>();
	@Input() public text = '';
	public hide() {
		this.isShow.set(false);
	}
}
