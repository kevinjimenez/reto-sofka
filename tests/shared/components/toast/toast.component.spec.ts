import { ComponentFixture, TestBed } from '@angular/core/testing';
import { ToastComponent } from '../../../../src/app/shared/components/toast/toast.component';

describe('ToastComponent', () => {
	let component: ToastComponent;
	let fixture: ComponentFixture<ToastComponent>;
	let compiled: HTMLElement;

	beforeEach(async () => {
		await TestBed.configureTestingModule({
			imports: [ToastComponent]
		}).compileComponents();
	});

	beforeEach(() => {
		fixture = TestBed.createComponent(ToastComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
		compiled = fixture.nativeElement;
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('should match snapshot', () => {
		expect(compiled).toMatchSnapshot();
	});

	it('should show the toast when show() is called', () => {
		component.isShow.set(true);
		expect(component.isShow()).toBe(true);
	});

	it('should hide the toast when hide() is called', () => {
		component.isShow();
		component.hide();
		expect(component.isShow()).toBe(false);
	});

	it('should display the text passed as input', () => {
		component.text = 'Test Toast Message';
		component.isShow.set(true);
		fixture.detectChanges();

		const toastContend = compiled.querySelector('.toast-contend');
		expect(toastContend?.textContent?.trim()).toBe(component.text.trim());

		// expect(component.text).toBe('Test Toast Message');
	});
});
