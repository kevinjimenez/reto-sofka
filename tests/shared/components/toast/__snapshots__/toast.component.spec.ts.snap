// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastComponent should match snapshot 1`] = `
<div
  id="root1"
>
  
</div>
`;
